CMSgov/easi-app
# [EASI-2486] Duplicate additional contacts bug

# EASI-2486

## Changes and Description

- First attempt to solve the bug was changing the `fetch-policy` on the contacts query in [src/hooks/useSystemIntakeContacts](https://github.com/CMSgov/easi-app/blob/EASI-2486/duplicate-additional-contacts-bug/src/hooks/useSystemIntakeContacts.tsx). The bug ended up not being a caching issue so this didn't fix anything, but I left the changes.
  - Changed query `fetch-policy` from `cache-first` to EASI's default `cache-and-network`
  - Removed conditional rendering based on query loading state
  - Updated Formik forms on all grt action views to reinitialize default values when contacts query is completed
  
- Updated method for displaying contacts to fix disconnect between new additional contacts and state values within `<Recipient />` component. State values weren't updating, so the previous contact was showing instead of the most recent contact added.
  -  Contacts are split into two different state arrays: `verifiedContacts` and `unverifiedContacts`
  - Initial unverified contacts stay in the `unverifiedContacts` array after verifying to preserve display order on rerender (initial unverified contacts stay above "show more" toggle).
  - Additional contacts are added to `verifiedContacts` array and now rendered in `<CheckboxField />` component instead of `<Recipient />`
  - These changes removed the `contactDetails` state that was causing the issue within the `<Recipient />` component
 


## How to test this change

I wasn't able to reliably reproduce the bug when testing - it only randomly happened twice when adding additional contacts. Testing all GRT actions and the contact details section of the system intake form should cover everything.

- GRT actions
  - Test on pre-seeded intakes so there are unverified contacts
  - Verify contacts
    - After verifying, contact should stay in same order and be automatically selected as recipient
  - Add additional contacts
    - Contact data should be correct
    - Contact should be automatically selected as recipient
- System intake form
  - Fill out contact details section required fields 
  - Test "business owner / product manager same as requester" checkbox
  - Add additional contacts
    - Edit contact
    - Delete contact

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
